dali/pipeline/data/tensor_vector.cc

```diff
@@ -766,22 +755,79 @@ TensorVector<Backend> &TensorVector<Backend>::operator=(TensorVector<Backend> &&
     tl_ = std::move(other.tl_);
     type_ = other.type_;
     sample_dim_ = other.sample_dim_;
-    views_count_ = other.views_count_.load();
     tensors_ = std::move(other.tensors_);
-    for (auto &t : tensors_) {
-      if (t) {
-        if (auto *del = std::get_deleter<ViewRefDeleter>(t->data_)) del->ref = &views_count_;
-      }
-    }
 
-    other.views_count_ = 0;
     other.curr_num_tensors_ = 0;
     other.tensors_.clear();
   }
   return *this;
 }
 
 
+// This is to check if we are actually laid down in contiguous memory
+// TODO(klecki): make this internal and name it something like: IsContiguouslyStored?
+template <typename Backend>
+bool TensorVector<Backend>::IsContiguousTensor() const {
```

Comment:
Actually, I'd swap the names, like:
IsContiguous -> UsesContiguousAllocation
this function -> IsContiguous

This would be consistent with TensorListView::is_contiguous which checks for incidental contiguity (not least because it doesn't manage the storage).

Reply:
We already have the SetContiguous/IsContiguous in several places, where it means the selected mode for the batch, making it UseContiguousAllocation is both quite a mouthful to type (and recently I reverted the change from `._num_elements()` to `.shape().num_elements()`) and changes the meaning of the API that is already used for different thing.

This is the only part of the ported API that actually isn't publicly exposed, so we can easily change though, hence my proposal.

Reply:
Maybe IsContiguousInMemory?